### PR TITLE
autotools and build recombobulation, plus minor code fiddling

### PR DESCRIPTION
**Pull request by barak**
This is a substantial rewrite/update of the build system, including modernization, getting various ./configure --enable/disable options to work, getting things to "make install" properly, getting "make dist" to include everything, using config.h more consistently, and the like. This includes a bit of fooling around with how the support files are dealt with. There is also a small amount of monkeying with the code proper: squelching warnings and removing ancient unneeded cases and such. (This started with the patches that were needed to get this to build across Debian architectures, and kind of grew from there.)

### CODE REVIEW

**Comment by jrincayc** [comment]
I have skimmed this and it looks to be useful.  I'll review it fully soon.  Thank you.

**Comment by jrincayc** [comment]
Hm, I would prefer if it checked for wx-config and if that was found default to --enable-wx.

File: .gitignore [hunk 1]
```diff
@@ -1,19 +1,28 @@
 *.o
-git.c
```
Comment (jrincayc) [inline .gitignore:2]: Hm, removing including the git version from the compile is a disadvantage.

File: Makefile.am [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+EXTRA_DIST += win32trm.c win32trm.h
+
+# IBM-specific graphics routines (unused)
+EXTRA_DIST += ztcterm.c ztcterm.h
+
+# Macintosh graphics routines (unused)
+EXTRA_DIST += macterm.c macterm.h
+
+# Ancient Makefile for Macintosh (unused)
+EXTRA_DIST += mac-makefile
```
Comment (jrincayc) [inline Makefile.am:73]: This and the corresponding file can probably be deleted at this point.

File: configure [hunk 3]
```diff
@@ -1,6104 +0,0 @@
-#! /bin/sh
```
Comment (jrincayc) [inline configure:1]: For what it is worth, leaving this in the repository simplifies building for me on macos.

File: configure.ac [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+AS_IF([test x${enable_objects} = xyes],
+ [AC_DEFINE([OBJECTS],[1],[enable object-oriented features])])
+AM_CONDITIONAL([OBJECTS], [test x${enable_objects} = xyes])
+
+AM_OPTIONS_WXCONFIG
+
+AC_MSG_CHECKING([enable_wx])
+AC_ARG_ENABLE([wx],
+  [AS_HELP_STRING([--enable-wx],[use wxWidgets (default=no)])],,
+  [enable_wx=no])
```
Comment (jrincayc) [inline configure.ac:32]: Hm, I would prefer that this checks if wx_config is available, and if so automatically enable it.  ./configure on linux should automatically find something to draw on or it won't be a very typical logo distribution.

File: logo.h [hunk 5]
```diff
@@ -70,7 +75,7 @@
 #endif
 #else
 #define RETSIGTYPE void
-#define SIGRET 
+#define SIGRET return 0;
```
Comment (jrincayc) [inline logo.h:78]: This line can just be eliminated since SIGRET is no-longer used.

File: main.c [hunk 6]
```diff
@@ -269,14 +269,10 @@ int main(int argc, char *argv[]) {
...
 #endif
       {
-#ifdef HAVE_WX
-	extern char *GIT;
-#endif
 	char version[64];
 	lcleartext(NIL);
 #ifdef HAVE_WX
 	strcpy(version,"6.1");
-	strcat(version,GIT);
```
Comment (jrincayc) [inline main.c:279]: I found having the git name in the version quite useful, but I am open to alternative ways to do this.

File: wxTerminal.cpp [hunk 7]
```diff
@@ -494,7 +498,13 @@ void LogoFrame::OnSave(wxCommandEvent& event) {
 void LogoFrame::OnSaveAs(wxCommandEvent& WXUNUSED(event)) {
 	wxFileDialog dialog(this,
 			    _T("Save Logo Workspace"),
-			    *wxEmptyString,
+			    (firstloadsave ?
+#ifdef __WXMAC__   /* needed for wxWidgets 2.6 */
```
Comment (jrincayc) [inline wxTerminal.cpp:502]: Mac is now using wxWidgets 3, so should not have this code.  Ditto for next diff.

**Comment by barak** [comment]
No objections, your points are all reasonable. I was rebasing some old tweaks, some of which (like removing the git.c business) were to get it to build on the Debian autobuilders. And I didn't do much cleaning, like the signal updates looked pretty crap.

Re including the git version: how about making that optional, `./configure --enable-git-version`?

**Comment by jrincayc** [comment]
> Re including the git version: how about making that optional, `./configure --enable-git-version`?

Okay, I can make the git version optional.

File: .gitignore (hunk 1 above)
Comment (jrincayc) [reply .gitignore:2]: Fixed in #46

File: Makefile.am (hunk 2 above)
Comment (jrincayc) [reply Makefile.am:73]: Fixed in #46

File: logo.h (hunk 5 above)
Comment (jrincayc) [reply logo.h:78]: Fixed in #46

File: main.c (hunk 6 above)
Comment (jrincayc) [reply main.c:279]: Fixed in #46

File: wxTerminal.cpp (hunk 7 above)
Comment (jrincayc) [reply wxTerminal.cpp:502]: Fixed in #46

File: configure.ac (hunk 4 above)
Comment (jrincayc) [reply configure.ac:32]: Fixed in #46

**Comment by jrincayc** [comment]
I have fixed the things I thought needed to be fix in #46  (I don't think I can upload changes to your branch directly).  I'll re-review it next week and merge both this one and the other one.

**Comment by jrincayc** [comment]
I tried building this on mac and realized there are several issues.  This may take longer than I expected.

**Pull request closed without merging**